test-utils/state-viewer/src/state_dump.rs

```diff
@@ -0,0 +1,201 @@
+use near_primitives::types::{AccountInfo, StateRoot};
+use near_store::TrieIterator;
+use neard::NightshadeRuntime;
+use std::collections::HashMap;
+
+pub fn state_dump(
+    runtime: NightshadeRuntime,
+    state_roots: Vec<StateRoot>,
+    last_block_header: BlockHeader,
+    genesis_config: &GenesisConfig,
+) -> Genesis {
+    println!("Generating genesis from state data");
+    let genesis_height = last_block_header.inner_lite.height + 1;
+    let block_producers = runtime
+        .get_epoch_block_producers_ordered(
+            &last_block_header.inner_lite.epoch_id,
+            &last_block_header.hash,
+        )
+        .unwrap();
+    let validators = block_producers
+        .into_iter()
+        .filter_map(|(info, is_slashed)| {
+            if !is_slashed {
+                Some((info.account_id, (info.public_key, info.stake)))
+            } else {
+                None
+            }
+        })
+        .collect::<HashMap<_, _>>();
+
+    let mut records = vec![];
+    for state_root in &state_roots {
+        let trie = TrieIterator::new(&runtime.trie, &state_root).unwrap();
+        for item in trie {
+            let (key, value) = item.unwrap();
+            if let Some(mut sr) = StateRecord::from_raw_key_value(key, value) {
+                if let StateRecord::Account { account_id, account } = &mut sr {
+                    if account.locked > 0 {
+                        if let Some((_, stake)) = validators.get(account_id) {
+                            account.amount = account.amount + account.locked - *stake;
+                            account.locked = *stake;
+                        } else {
+                            account.amount += account.locked;
+                            account.locked = 0;
+                        }
+                    }
+                }
+                records.push(sr);
+            }
+        }
+    }
+
+    let mut genesis_config = genesis_config.clone();
+    genesis_config.genesis_height = genesis_height;
+    genesis_config.validators = validators
+        .into_iter()
+        .map(|(account_id, (public_key, amount))| AccountInfo { account_id, public_key, amount })
+        .collect();
+    Genesis::new(genesis_config, records.into())
+}
+
+#[cfg(test)]
+mod test {
+    use crate::state_dump::state_dump;
+    use near_chain::{ChainGenesis, RuntimeAdapter};
+    use near_chain_configs::Genesis;
+    use near_client::test_utils::TestEnv;
+    use near_crypto::{InMemorySigner, KeyType};
+    use near_primitives::transaction::SignedTransaction;
+    use near_primitives::types::NumBlocks;
+    use near_store::test_utils::create_test_store;
+    use near_store::Store;
+    use neard::config::GenesisExt;
+    use neard::config::TESTING_INIT_STAKE;
+    use neard::genesis_validate::validate_genesis;
+    use neard::NightshadeRuntime;
+    use std::collections::HashSet;
+    use std::iter::FromIterator;
+    use std::path::Path;
+    use std::sync::Arc;
```

Comment:
... I know you guys hate grouping the imports with optimize imports, but crate, then internal deps then stds just looks weird ...